pkg/controller/webhookconfigurationcabundle/controller/webhookconfiguration_injecting_controller.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	admissionregistrationinformers "k8s.io/client-go/informers/admissionregistration/v1beta1"
+	admissionregistrationclient "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	listers "k8s.io/client-go/listers/admissionregistration/v1beta1"
+	"k8s.io/klog"
+
+	"github.com/openshift/service-ca-operator/pkg/boilerplate/controller"
+	"github.com/openshift/service-ca-operator/pkg/controller/api"
+)
+
+type webhookConfigurationCABundleInjetorController struct {
+	configMapClient admissionregistrationclient.ValidatingWebhookConfigurationsGetter
+	configMapLister listers.ValidatingWebhookConfigurationLister
+
+	caBundle []byte
+}
+
+func NewValidatingWebhookConfigurationCABundleInjectionController(webhookConfigurationInformer admissionregistrationinformers.ValidatingWebhookConfigurationInformer, webhookConfigurationClient admissionregistrationclient.ValidatingWebhookConfigurationsGetter, caBundle []byte) controller.Runner {
+	ic := &webhookConfigurationCABundleInjetorController{
+		configMapClient: webhookConfigurationClient,
+		configMapLister: webhookConfigurationInformer.Lister(),
+		caBundle:        caBundle,
+	}
+
+	return controller.New("ValidatingWebhookConfigurationCABundleInjectionController", ic,
+		controller.WithInformer(webhookConfigurationInformer, controller.FilterFuncs{
+			AddFunc:    api.HasInjectCABundleAnnotation,
+			UpdateFunc: api.HasInjectCABundleAnnotationUpdate,
+		}),
+	)
+}
+
+// NOTE: ValidatingWebhookConfiguration is nonNamespaced
+func (ic *webhookConfigurationCABundleInjetorController) Key(namespace, name string) (metav1.Object, error) {
+	return ic.configMapLister.Get(name)
+}
+
+func (ic *webhookConfigurationCABundleInjetorController) Sync(obj metav1.Object) error {
+	sharedValidatingWebhookConfiguration := obj.(*admissionregistrationv1beta1.ValidatingWebhookConfiguration)
+
+	// check if we need to do anything
+	if !api.HasInjectCABundleAnnotation(sharedValidatingWebhookConfiguration) {
+		return nil
+	}
+
+	// ensure data of validatingwebhookconfiguration
+	return ic.ensureValidatingWebhookConfigurationCABundleInjection(sharedValidatingWebhookConfiguration)
+}
+
+func (ic *webhookConfigurationCABundleInjetorController) ensureValidatingWebhookConfigurationCABundleInjection(current *admissionregistrationv1beta1.ValidatingWebhookConfiguration) error {
+	// skip updating when the CA bundle is already there
```

Comment:
The client CA bundle can be rotated, so you will need to be sure to allow for it to update.

Reply:
Thanks for the hint, I'll consider this.

Reply:
I removed the check for the moment as you suggested.